core/CronArchive.php

```diff
@@ -277,15 +277,16 @@ public function init()
         $websitesIds = $this->initWebsiteIds();
         $this->filterWebsiteIds($websitesIds);
 
-        if (!empty($this->shouldArchiveSpecifiedSites)
-            || !empty($this->shouldArchiveAllSites)
-            || !SharedSiteIds::isSupported()) {
+        if (!empty($this->shouldArchiveAllSites) && SharedSiteIds::isSupported()) {
+            $this->websites = new SharedSiteIds($websitesIds, SharedSiteIds::OPTION_ALL_WEBSITES);
+        } elseif (!empty($this->shouldArchiveSpecifiedSites) || !SharedSiteIds::isSupported()) {
```

Comment:
A suggestion to improve clarity (at least for this small part of CronArchive.php): could the site ID list creation logic be in a private method? ie `$this->websites = $this->createSitesToArchiveQueue(...);` then instead of an if/elseif, could have several if(s) w/ return statements.